driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -452,14 +456,15 @@ private void notifyOwnerWhenDefunct() {
     }
 
     String keyspace() {
-        return keyspace;
+        SetKeyspaceAttempt currentAttempt = targetKeyspace.get();
+        return currentAttempt == null ? null : currentAttempt.keyspace;
```

Comment:
sgtm. There's no way to set the keyspace back to null, so resetting to `defaultKeyspaceAttempt` guarantees that the call will be retried.